graphql/graphql.go

```diff
@@ -285,6 +286,34 @@ func (t *Transaction) MaxPriorityFeePerGas(ctx context.Context) (*hexutil.Big, e
 	}
 }
 
+func (t *Transaction) EffectiveTip(ctx context.Context) (*hexutil.Big, error) {
+	tx, err := t.resolve(ctx)
+	if err != nil || tx == nil {
+		return nil, err
+	}
+	header, err := t.block.resolveHeader(ctx)
+	if err != nil || header == nil {
+		return nil, err
+	}
+	if header.BaseFee == nil {
+		return (*hexutil.Big)(tx.GasPrice()), nil
+	}
+
+	// TODO: What to return for non-dynamicfee tx types?
```

Comment:
`gasprice - basefee`

Reply:
This should work: `tx.EffectiveGasTip(header.BaseFee)` -- please test if it doesn't. Would be interesting to know why it would not work